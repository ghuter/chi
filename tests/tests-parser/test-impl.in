A :: impl X {
	f :: fun();
}

B :: impl X[T] {
	f :: fun();
}

C :: impl X[K, V] {
	f :: fun();
}

D :: impl X[K, V] (M: Module) {
	f :: fun();
}

E :: impl X[K, V] (M: Module[K,V]) {
	f :: fun();
}
