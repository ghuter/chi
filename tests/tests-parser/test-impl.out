SMODIMPL(A, [], (), {SSIGN(f) : void}) : X
SMODIMPL(B, [T], (), {SSIGN(f) : void}) : X
SMODIMPL(C, [K, V], (), {SSIGN(f) : void}) : X
SMODIMPL(D, [K, V], (M: Module[]), {SSIGN(f) : void}) : X
SMODIMPL(E, [K, V], (M: Module[K, V]), {SSIGN(f) : void}) : X
